plugin/core/sessions.py

```diff
@@ -1745,13 +1758,22 @@ def m_workspace_inlayHint_refresh(self, params: None, request_id: Any) -> None:
         for sv in not_visible_session_views:
             sv.session_buffer.set_inlay_hints_pending_refresh()
 
+    def m_workspace_diagnostic_refresh(self, params: None, request_id: Any) -> None:
+        """handles the workspace/diagnostic/refresh request"""
+        self.send_response(Response(request_id, None))
```

Comment:
It's perhaps a bit unclear whether the response should be sent after or before the related requests are finished but I guess we are currently consistent with ourselves at least.

Reply:
I would think that if it was expected to postpone the response until the other requests were sent, then this would explictily be mentioned in the specs. So I think the requests are independent and it should be fine to immediately send the response here.